src/System.Linq/src/System/Linq/Select.cs

```diff
@@ -852,7 +917,18 @@ public List<TResult> ToList()
 
             public int GetCount(bool onlyIfCheap)
             {
-                return Count;
+                int count = Count;
+
+                if (!onlyIfCheap)
+                {
+                    int end = _minIndexInclusive + count;
+                    for (int i = _minIndexInclusive; i != end; ++i)
```

Comment:
An issue came up here that I wasn't sure best how to approach:

- `lazyEnumerable.Select(i => i).Skip(1).Count()` runs the selector `lazyEnumerable.Count()` times, because `Select.Skip` on a lazy enumerable isn't specially recognized and the selector gets run on the first time.

- `list.Select(i => i).Skip(1).Count()` is specially recognized, however, and it returns a `SelectListPartitionIterator` which does not run the selector on the first item.

One way to fix this would be to start from `0` instead of `_minIndexInclusive` here. However, if we do that, we break `Skip(1).Select(i => i)` which also ends up here; patterns like those should definitely not run the selector on the first item.

Ideally, we would somehow have a way to differentiate if `Skip` or `Select` was called first from within the iterator, and start from `_minIndexInclusive` or `0` accordingly. But then we might need to add an extra field...

cc @JonHanna